lib/case.js

```diff
@@ -47,13 +48,13 @@ module.exports = (metatests) => {
         expected = definition.pop();
         expectedType = typeof(expected);
         if (expectedType === 'function') sExpected = 'function';
-        else if (expectedType === 'string') sExpected = '"' + expected + '"';
-        else sExpected = JSON.stringify(expected);
+        else if (expectedType === 'string') sExpected = '\'' + expected + '\'';
+        else sExpected = inspect(expected);
```

Comment:
This won't work for nested objects with nesting depth > 2 since that's the default `util.inspect()` behavior.

Reply:
Example of the test that fails properly on `master` and isn't failing in this PR's branch:
```js
const f = x => x;
metatests.namespace({ f });
metatests.case('Failing test', {
  f: [
    [{ a: { b: { c: { d: 10 } } } }, { a: { b: { c: { d: {} } } } }],
  ],
});
```

Reply:
Also, could you suggest a test to check this as I couldn't come up with one? (Without `depth` it will just print i.e. `Array` instead of actual values, therefore, we will be getting success even though objects may differ but is there a way to test this without 'negative-declarative-tests'?).

Reply:
@lundibundi you can set `depth` option to `null`, that way `inspect` will recurse the objects indefinitely (I am not sure if it's safe for circular references though).

Reply:
@lundibundi, ah, sorry, I missed that. I am not sure how we can test that, except for some kind of inverted declarative test if such functionality is implemented here, which I am not sure of.